plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/AppEngineLibraryContainerInitializer.java

```diff
@@ -0,0 +1,54 @@
+package com.google.cloud.tools.eclipse.appengine.libraries;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IConfigurationElement;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.jdt.core.IClasspathContainer;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.JavaCore;
+
+import com.google.cloud.tools.eclipse.appengine.libraries.config.LibraryBuilder;
+import com.google.common.annotations.VisibleForTesting;
+
+public class AppEngineLibraryContainerInitializer extends org.eclipse.jdt.core.ClasspathContainerInitializer {
+
+  private static final String LIBRARIES_EXTENSION_POINT = "com.google.cloud.tools.eclipse.appengine.libraries";
+
+  private Map<String, Library> libraries;
+
+  public AppEngineLibraryContainerInitializer() {
+    super();
+    IConfigurationElement[] configurationElements = Platform.getExtensionRegistry().getConfigurationElementsFor(LIBRARIES_EXTENSION_POINT);
```

Comment:
So personally I prefer to make passive constructors and instead have this sort of stuff done in `initialize()`.  It can be difficult to figure out what's gone wrong  with exceptions in a constructor.